views/layout.erb

```diff
@@ -119,14 +40,15 @@ li:last-child > a:focus {
   </nav>
 </header>
 
-<%# FIXME: add alerts for if an error occurs; show popen instead %>
-
 <div class="container" role="main">
+  <% if @error %>
+      <div class="alert alert-danger"><%= @error %></div>
+  <% end %>
+
   <%= yield %>
 </div>
 
-<%# FIXME: replace with local built bootstrap & jquery in /public/ %>
-<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script>
-<script src="https://code.jquery.com/jquery-3.2.1.min.js" integrity="sha256-hwg4gsxgFZhOsEEamdOYGBf13FyQuiTwlAQgxVSNgt4=" crossorigin="anonymous"></script>
+<script src="<%= url("/bootstrap-3.3.7-dist/js/bootstrap.min.js") %>"></script>
```

Comment:
https://getbootstrap.com/docs/3.3/javascript/

>Plugins can be included individually (using Bootstrap's individual *.js files), or all at once (using bootstrap.js or the minified bootstrap.min.js).

Which also means if none of the plugins are being used, all of it can be omitted.